test/torchaudio_unittest/prototype/hdemucs_test_impl.py

```diff
@@ -0,0 +1,76 @@
+import torch
+from parameterized import parameterized
+from torchaudio.prototype.models.hdemucs import HDemucs, _HEncLayer, _HDecLayer
+from torchaudio_unittest.common_utils import TestBaseMixin, torch_script
+
+_SOURCE_SETS = [
+    (["bass", "drums", "other", "vocals"],),
+    (["bass", "drums", "other"],),
+    (["bass", "vocals"],),
+    (["vocals"],),
+]
+
+
+def _get_hdemucs_model(sources):
+    return HDemucs(sources)
+
+
+class TorchscriptConsistencyMixin(TestBaseMixin):
+    r"""Mixin to provide easy access assert torchscript consistency"""
+
+    def _assert_torchscript_consistency(self, model, tensors):
+        ts_func = torch_script(model)
+
+        torch.random.manual_seed(40)
+        output = model(*tensors)
+
+        torch.random.manual_seed(40)
+        ts_output = ts_func(*tensors)
+
+        self.assertEqual(ts_output, output)
+
+
+class HDemucsTests(TorchscriptConsistencyMixin):
+    def _get_inputs(self, length: int):
+        sample = torch.rand(1, 2, length * 44100, dtype=torch.float32, device=self.device)
+        return sample
+
+    @parameterized.expand(_SOURCE_SETS)
+    def test_hdemucs_output_shape(self, sources):
+        r"""Feed tensors with specific shape to HDemucs and validate
+        that it outputs with a tensor with expected shape.
+        """
+        length = 10
+
+        model = _get_hdemucs_model(sources).to(self.device).eval()
+        inputs = self._get_inputs(length)
+
+        split_sample = model(inputs)
+
+        assert split_sample.shape == (1, len(sources), 2, length * 44100)
+
+    def test_encoder_output_shape(self):
+        r"""Feed tensors with specific shape to HDemucs Decoder and validate
+        that it outputs with a tensor with expected shape.
+        """
+        chin, chout = 4, 48
+        model = _HEncLayer(chin, chout).to(self.device).eval()
+
+        x = torch.rand(1, chin, 2048, 739, device=self.device, dtype=self.dtype)
```

Comment:
can you make variables similar to `chin`, `chout` to indicate what 1, 2048,  and 739 represent? and if you could reuse them in the result that makes it more intuitive to understand since these may look like arbitrary numbers currently